linera-sdk/src/views/system_api.rs

```diff
@@ -37,28 +45,41 @@ impl KeyValueStoreClient for KeyValueStore {
     type Keys = Vec<Vec<u8>>;
     type KeyValues = Vec<(Vec<u8>, Vec<u8>)>;
 
+    fn max_key_size(&self) -> usize {
+        MAX_KEY_SIZE
+    }
+
     fn max_stream_queries(&self) -> usize {
         1
     }
 
-    async fn read_key_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
-        let future = wit::ReadKeyBytes::new(key);
+    async fn read_value_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
+        if key.len() > MAX_KEY_SIZE {
+            return Err(ViewError::KeyTooLong);
+        }
+        let future = wit::ReadValueBytes::new(key);
         Ok(future::poll_fn(|_context| future.poll().into()).await)
     }
 
-    async fn read_multi_key_bytes(
+    async fn read_multi_values_bytes(
         &self,
         keys: Vec<Vec<u8>>,
     ) -> Result<Vec<Option<Vec<u8>>>, Self::Error> {
         let mut results = Vec::new();
         for key in keys {
-            let value = self.read_key_bytes(&key).await?;
+            if key.len() > MAX_KEY_SIZE {
+                return Err(ViewError::KeyTooLong);
+            }
```

Comment:
If rustfmt leaves it as a single line, `ensure!` (from `linera-base`) might be more concise.

```suggestion
            ensure!(key.len() <= MAX_KEY_SIZE, ViewError::KeyTooLong);
```

Reply:
Ok, done.